src/AutoRest.CSharp/Mgmt/Generation/ReferenceTypeWriter.cs

```diff
@@ -15,23 +15,30 @@ protected override void AddClassAttributes(CodeWriter writer, ObjectType objectT
         {
             if (objectType is not SchemaObjectType schema)
                 return;
-            var extensions = schema.ObjectSchema.Extensions;
-            if (extensions != null)
+
+            if (MgmtReferenceType.IsPropertyReferenceType(schema.ObjectSchema))
             {
                 if (Configuration.IsBranded)
-                    writer.UseNamespace("Azure.Core");
-                if (extensions.MgmtReferenceType)
                 {
-                    writer.Line($"[{ReferenceClassFinder.ReferenceTypeAttribute}]");
+                    writer.UseNamespace("Azure.Core");
                 }
-                else if (extensions.MgmtPropertyReferenceType)
+                writer.Line($"[{ReferenceClassFinder.PropertyReferenceTypeAttribute}]");
+            }
+            else if (MgmtReferenceType.IsTypeReferenceType(schema.ObjectSchema))
+            {
+                if (Configuration.IsBranded)
                 {
-                    writer.Line($"[{ReferenceClassFinder.PropertyReferenceTypeAttribute}]");
+                    writer.UseNamespace("Azure.Core");
                 }
-                else if (extensions.MgmtTypeReferenceType)
+                writer.Line($"[{ReferenceClassFinder.TypeReferenceTypeAttribute}]");
+            }
+            else if (MgmtReferenceType.IsReferenceType(schema.ObjectSchema))
+            {
+                if (Configuration.IsBranded)
                 {
-                    writer.Line($"[{ReferenceClassFinder.TypeReferenceTypeAttribute}]");
+                    writer.UseNamespace("Azure.Core");
                 }
+                writer.Line($"[{ReferenceClassFinder.ReferenceTypeAttribute}(new string[]{{\"SystemData\"}})]");
```

Comment:
well we are hard coding a property name here which might not work if we would like to extend in the future.
But I do not think it is going to happen in any near or far future, so maybe we could add a TODO item here or add some comments here to note this hard coding because this only works for ResourceData and trackedResourceData case.

Reply:
Added comments